python/foglamp/services/south/server.py

```diff
@@ -213,16 +184,45 @@ def run(self):
                 lambda: asyncio.ensure_future(self._stop(loop)))
 
         asyncio.ensure_future(self._start(loop))
+
+        # This activates event loop and starts fetching events to the microservice server instance
         loop.run_forever()
 
+    async def _stop(self, loop):
+        if self._plugin is not None:
+            try:
+                self._plugin.plugin_shutdown(self._plugin_handle)
+            except Exception:
+                _LOGGER.exception("Unable to stop plugin '{}'".format(self._name))
+            finally:
+                self._plugin = None
+                self._plugin_handle = None
+
+        try:
+            await Ingest.stop()
+        except Exception:
+            _LOGGER.exception('Unable to stop the Ingest server')
+            return
+
+        # Finish all pending asyncio tasks
+        pending = asyncio.Task.all_tasks()
+        asyncio.gather(*pending)
+
+        # This deactivates event loop and helps aiohttp microservice server instance in graceful shutdown
+        loop.stop()
+
     async def shutdown(self, request):
         """implementation of abstract method form foglamp.common.microservice.
         """
-        print("shutdown south")
-        return web.json_response({"south":"shutdown"})
+        _LOGGER.info('Stopping South Service plugin {}'.format(self._name))
```

Comment:
let's wrap this into `try` `except`; in `except Exception as ex:` `return web.HTTPException(reason=str(ex))`